lib/TPP/ConvertCheckToLoops.cpp

```diff
@@ -107,8 +106,70 @@ struct ConvertExpectTrueOp : public OpRewritePattern<ExpectTrueOp> {
   }
 };
 
+// Convert
+// check.expect_sane(%t1:tensor<2x2xf32>)
+// to the pattern:
+// scf.some_loop(%i, %j)
+// %0 = load from %t1
+// %1 = arith.cmpf ord, %0, %nan : f32
+// %2 = arith.cmpf one, %0, %inf : f32
+// %compare = arith.andi %1, %2 : i1
+// cf.assert %compare, "Buffer can't contain NaNs or Infinite values"
+struct ConvertExpectSaneOp : public OpRewritePattern<ExpectSaneOp> {
+  using OpRewritePattern<ExpectSaneOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(ExpectSaneOp expectSaneOp,
+                                PatternRewriter &rewriter) const override {
+    Location loc = expectSaneOp.getLoc();
+    auto elementType = expectSaneOp.getOperand().getType().getElementType();
```

Comment:
Nit: cache operand, type and element type to use in the code below, makes for clearer code.